src/http/ws_rpc_context.h

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "../ds/serialized.h"
+#include "enclave/rpc_context.h"
+#include "http_parser.h"
+#include "http_sig.h"
+
+namespace ws
+{
+  static std::vector<uint8_t> serialise(
+    size_t code,
+    const std::vector<uint8_t>& body,
+    kv::Version commit = 0,
+    kv::Consensus::View term = 0,
+    kv::Version global_commit = 0)
+  {
+    size_t frame_size = ws::OUT_CCF_HEADER_SIZE + body.size();
+    size_t sz_size = 0;
+    if (frame_size > 125)
+    {
+      sz_size = frame_size > std::numeric_limits<uint16_t>::max() ? 8 : 2;
+    }
+
+    size_t ws_h_size = ws::INITIAL_READ + sz_size;
+    std::vector<uint8_t> msg(ws_h_size + frame_size);
+    msg[0] = 0x82;
+    switch (sz_size)
+    {
+      case 0:
+      {
+        msg[1] = frame_size;
+        break;
+      }
+      case 2:
+      {
+        msg[1] = 0x7e;
+        *((uint16_t*)&msg[2]) = htons(frame_size);
+        break;
+      }
+      case 8:
+      {
+        msg[1] = 0x7f;
+        *((uint64_t*)&msg[2]) = htobe64(frame_size);
+        break;
+      }
+      default:
+        throw std::logic_error("Unreachable");
```

Comment:
Error message is rather vague here. Perhaps mention the `sz_size` here too?

Reply:
sz_size is 0 | 2 | 8 though (see line 20), this really is unreachable.